packages/kubernetes-1.20/kubelet-exec-start-conf

```diff
@@ -17,6 +17,12 @@ ExecStart=/usr/bin/kubelet \
     --network-plugin cni \
     --root-dir /var/lib/kubelet \
     --cert-dir /var/lib/kubelet/pki \
+{{#if settings.kubernetes.credential-providers}}
+{{#if (any-enabled settings.kubernetes.credential-providers)}}
+    --image-credential-provider-bin-dir /usr/libexec/kubernetes/kubelet/plugins \
+    --image-credential-provider-config /etc/kubernetes/kubelet/credential-provider-config.yaml \
+{{/if}}
+{{/if}}
```

Comment:
Good call! This was originally needed when the structure was `settings.kubernetes.credential-providers.providers` since credential-providers would be null and cause an error. But now that we've simplified the structure I don't think there is any reason to have the extra check.

Reply:
Turns out there actually was a reason I did it this way. If there is not a check before passing in the the handler there is an error rendering the file:

```
Error rendering "kubelet-exec-start-conf" line 20, col 1: `any_enabled` helper: Couldn't read parameter arg
```

I tried changing it to:

```
{{#if (and settings.kubernetes.credential-providers (any-enabled settings.kubernetes.credential-providers))}}
```

But it appears handlebars doesn't shortcircuit evaluation, so even though the first condition evaluates to falsey, it still tries to run the second part.